### PR TITLE
Add print/plot S3 methods for VennDiagram

### LINKED ISSUE #26: Implement print/plot S3 methods for VennDiagram
If `filename = NULL` then the `gList` object representing the plot is returned. The default `gList` print message is quite cryptic

``` r
library(VennDiagram)
venn.plot <- venn.diagram(
    list(A = 1:150, B = 121:170), 
    filename = NULL,
    disable.logging = TRUE
    )

venn.plot
#> (polygon[GRID.polygon.1], polygon[GRID.polygon.2], polygon[GRID.polygon.3], polygon[GRID.polygon.4], text[GRID.text.5], text[GRID.text.6], text[GRID.text.7], text[GRID.text.8], text[GRID.text.9])
```

<sup>Created on 2023-01-12 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

Adding a `VennDiagram` S3 class to the returned object will enable implementing `print` and `plot` functions to display the plot in an interactive session. This will not change the ability to manipulate the `gList` but will enable easier interactive use of VennDiagram. The is also how we implement in BPG: https://github.com/uclahs-cds/public-R-BoutrosLab-plotting-general/blob/main/R/create.multipanelplot.R#L1-L24 and other common packages such as ggplot2: https://github.com/tidyverse/ggplot2/blob/HEAD/R/plot.r#L162-L190 as well as lattice.

## Desired outcome:

``` r
library(VennDiagram)
venn.plot <- venn.diagram(
    list(A = 1:150, B = 121:170), 
    filename = NULL,
    disable.logging = TRUE
    );

venn.plot;
```

<sup>Created on 2023-01-12 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

### PR DESCRIPTION
**Pull request by stefaneng**
# Description
Implement print/plot S3 methods for VennDiagram class.

### Closes #26 

# Checklist

- [x] This PR does NOT contain [PHI](https://ohrpp.research.ucla.edu/hipaa/) or germline genetic data.  A repo may need to be deleted if such data is uploaded. Disclosing PHI is a [major problem](https://healthitsecurity.com/news/ucla-health-reaches-7.5m-settlement-over-2015-breach-of-4.5m).
- [x] This PR does NOT contain molecular files, compressed files, output files such as images (*e.g.* `.png`, .`jpeg`), `.pdf`, `.RData`, `.xlsx`, `.doc`, `.ppt`, or other non-plain-text files.  To automatically exclude such files using a [.gitignore](https://docs.github.com/en/get-started/getting-started-with-git/ignoring-files) file, see [here](https://github.com/uclahs-cds/template-base/blob/main/.gitignore) for example.
- [x] I have read the [code review guidelines](https://confluence.mednet.ucla.edu/display/BOUTROSLAB/Code+Review+Guidelines) and the [code review best practice on GitHub check-list](https://confluence.mednet.ucla.edu/display/BOUTROSLAB/Code+Review+Best+Practice+on+GitHub+-+Check+List).
- [x] I have set up or verified the `main` branch protection rule following the [github standards](https://confluence.mednet.ucla.edu/pages/viewpage.action?spaceKey=BOUTROSLAB&title=GitHub+Standards#GitHubStandards-Branchprotectionrule) before opening this pull request.
- [x] The name of the branch is meaningful and well formatted following the [standards](https://confluence.mednet.ucla.edu/display/BOUTROSLAB/Code+Review+Best+Practice+on+GitHub+-+Check+List), using [AD_username (or 5 letters of AD if AD is too long)]-[brief_description_of_branch].
- [x] I have added the major changes included in this pull request to the `CHANGELOG.md` under the next release version or unreleased, and updated the date.

## Example

No `grid.draw` call required.

``` r
library(VennDiagram)
VennDiagram::venn.diagram(list(A = 1:150, B = 121:170), filename = NULL)
```

### CODE REVIEW

File: NEWS [hunk 1]
```diff
@@ -1,3 +1,8 @@
+VennDiagram 1.7.4 2023-02-28 (Stefan Eng)
```
Comment (dan-knight) [inline NEWS:1]: Should be `1.8.0`

File: DESCRIPTION [hunk 2]
```diff
@@ -1,5 +1,5 @@
 Package: VennDiagram
-Version: 1.7.4
+Version: 1.7.5
```
Comment (dan-knight) [inline DESCRIPTION:2]: Should this be `1.8.0` as this is technically new functionality?

**Comment by dan-knight** [comment]
Very cool quality of life improvement. Could we add some unit tests for the different cases in the S3 function?

**Comment by stefaneng** [comment]
> Very cool quality of life improvement. Could we add some unit tests for the different cases in the S3 function?

I'm not quite sure how to test this.

**Comment by dan-knight** [comment]
> I'm not quite sure how to test this. Any suggestions?

Given the current end-to-end test cases, do the changes you made here affect them as written? In other words, if this code was incorrect, would it make the existing test cases fail?

**Comment by stefaneng** [comment]
> > I'm not quite sure how to test this. Any suggestions?
> 
> Given the current end-to-end test cases, do the changes you made here affect them as written? In other words, if this code was incorrect, would it make the existing test cases fail?

All the old tests still work. No, it would not make the tests fail since there old tests don't look at class at all and there is no print or plot method called at all.

**Comment by dan-knight** [comment]
> No, it would not make the tests fail since there old tests don't look at class at all and there is no print or plot method called at all.

Right. It might just be good to have some code that explicitly calls the S3 methods you defined here. Also, maybe a test case that asserts `is('VennDiagram')`?

**Review by dan-knight — APPROVED** [review]

**Pull request merged**